awx_collection/plugins/module_utils/tower_module.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.six import PY2, string_types
+from ansible.module_utils.six.moves import StringIO
+from ansible.module_utils.six.moves.urllib.parse import urlparse
+from ansible.module_utils.six.moves.configparser import ConfigParser, NoOptionError
+from socket import gethostbyname
+import re
+from os.path import isfile, expanduser, split, join, exists, isdir
+from os import access, R_OK, getcwd
+from distutils.util import strtobool
+
+try:
+    import yaml
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+
+class ConfigFileException(Exception):
+    pass
+
+
+class ItemNotDefined(Exception):
+    pass
+
+
+class TowerModule(AnsibleModule):
```

Comment:
It doesn't make a whole lot of sense to me that the common `TowerModule` would subclass `AnsibleModule`. Because of this, we still maintain the hack of the `direct_params` in `__init__`. I thought getting rid of that would kind of be the whole point of separating the classes.

The way it is now, it looks like you could have avoided moving all this code and just had the towerkit class subclass the old class.

But maybe this is your objective and you just haven't gotten around to everything yet.

Reply:
Correct, there is a plan in place but it will not be part of this PR.